lib/opal/nodes/literal.rb

```diff
@@ -156,7 +156,15 @@ def compile_static_regexp
           message = "named captures are not supported in javascript: #{value.inspect}"
           push "self.$raise(new SyntaxError('#{message}'))"
         else
-          push "#{Regexp.new(value).inspect}#{flags.join}"
+          regexp = Regexp.new(value)
+          flags = self.flags
+
+          if flags && flags !~ /^[gim]*$/
+            compiler.warning "Ignoring unsupported flags #{flags.inspect} found for regexp #{regexp.inspect}"
```

Comment:
Nowadays JS supports some more flags. For example: u. (All supported: dgimsuy).

In Ruby `u` is set by default, denotes the UTF-8 regexp encoding, in JS `u` enables the \p{..} and treats the string as UTF-16 and not UCS-2 (ie. support for characters 0x010000 - 0x10ffff, eg. emoji).

From what I know, this is used in Asciidoctor for instance.